pyomo/contrib/alternative_solutions/balas.py

```diff
@@ -71,16 +72,21 @@ def enumerate_binary_solutions(
         Boolean indicating that the solver output should be displayed.
     seed : int
         Optional integer seed for the numpy random number generator
+    pool_manager : None
+        Optional pool manager that will be used to collect solutions
 
     Returns
     -------
-    solutions
-        A list of Solution objects.
-        [Solution]
+    pool_manager
+        A PyomoPoolManager object
 
     """
     logger.info("STARTING NO-GOOD CUT ANALYSIS")
 
+    assert num_solutions >= 1, "num_solutions must be positive integer"
```

Comment:
Argument checking of public APIs should raise exceptions and not use assertions (running python in optimized mode will skip *all* assertions).

Reply:
Done (except for a few assertions buried in old code that I'm not sure how to rework).